internal/config/config.go

```diff
@@ -39,13 +40,14 @@ type Complement struct {
 	CAPrivateKey  *rsa.PrivateKey
 }
 
+var hsRegex = regexp.MustCompile(`COMPLEMENT_BASE_IMAGE_HS\d+=.+$`)
```

Comment:
Not all tests use the names of the form `hsN` - it isn't part of the public API and tests are free to use whatever names they want. As a result, we need to be more flexible here. HS names _are_ case insensitive as they form the domain part of the server, so it would be better to grab all env vars matching `COMPLEMENT_BASE_IMAGE_$HSNAME=$IMAGE_NAME` rather than hardcoding `HS_`.